tuf/data/types.go

```diff
@@ -181,6 +182,34 @@ type FileMeta struct {
 	Custom *json.RawMessage `json:"custom,omitempty"`
 }
 
+// Equals returns true if the other FileMeta object is equivalent to this one
+func (f FileMeta) Equals(o FileMeta) bool {
+	if o.Length != f.Length || len(f.Hashes) != len(f.Hashes) {
+		return false
+	}
+	if f.Custom == nil && o.Custom != nil || f.Custom != nil && o.Custom == nil {
```

Comment:
If `f.Custom` is `nil`, won't we get a panic trying to get `*f.Custom`?  Also, if neither were nil, I think that still wouldn't work because the equality operator doesn't work on byte slices.  :|  For instance:  https://play.golang.org/p/jY4yrw7lrL